src/internal/account.ts

```diff
@@ -404,16 +405,39 @@ export async function getAccountTransactionsCount(args: {
 export async function getAccountCoinAmount(args: {
   aptosConfig: AptosConfig;
   accountAddress: AccountAddressInput;
-  coinType: MoveStructId;
+  coinType?: MoveStructId;
+  faMetadataAddress?: AccountAddressInput;
 }): Promise<number> {
-  const { aptosConfig, accountAddress, coinType } = args;
+  const { aptosConfig, accountAddress, coinType, faMetadataAddress } = args;
+  let faAddress: string;
+
+  if (coinType !== undefined && faMetadataAddress !== undefined) {
+    faAddress = AccountAddress.from(faMetadataAddress).toStringLong();
+  } else if (coinType !== undefined && faMetadataAddress === undefined) {
+    // TODO Move to a separate function as defined in the AIP for coin migration
+    if (args.coinType === APTOS_COIN) {
+      faAddress = AccountAddress.A.toStringLong();
+    } else {
+      faAddress = createObjectAddress(AccountAddress.A, coinType).toStringLong();
+    }
+  } else if (coinType === undefined && faMetadataAddress !== undefined) {
+    faAddress = AccountAddress.from(faMetadataAddress).toStringLong();
```

Comment:
If faMetadataAddress is set, should this convert to the coinType?

Reply:
There's no way to retrieve the coin type from the FA metadata address.

So, in this case I'm taking it as an FA by itself.

Reply:
This is the expected behavior.